### ISSUE TITLE
Large file byte ops position references don't work?

### ISSUE THREAD

**Issue by kitsonk** [denoland/deno#14196]
A user of oak (@mdkq oakserver/oak#503) has reported an issue when uploading large file via oak. The user is getting a JavaScript stack trace like this:

```
[uncaught application error]: TypeError - out of range integral type conversion attempted

request: { url: "http://localhost/api/fragment", method: "POST", hasBody: true }
response: { status: 404, type: undefined, hasBody: false, writable: true }

    at async ftruncate (deno:runtime/js/30_fs.js:258:5)
    at async parts (https://deno.land/x/oak@v10.5.1/multipart.ts:253:15)
    at async FormDataReader.read (https://deno.land/x/oak@v10.5.1/multipart.ts:415:15)
    at async post (file:///Users/marcus/Desktop/rose/server/controllers/fragment.js:22:18)
    at async dispatch (https://deno.land/x/oak@v10.5.1/middleware.ts:41:7)
    at async dispatch (https://deno.land/x/oak@v10.5.1/middleware.ts:41:7)
    at async restrict (file:///Users/marcus/Desktop/rose/server/middleware/auth.js:20:25)
    at async dispatch (https://deno.land/x/oak@v10.5.1/middleware.ts:41:7)
    at async dispatch (https://deno.land/x/oak@v10.5.1/middleware.ts:41:7)
    at async dispatch (https://deno.land/x/oak@v10.5.1/middleware.ts:41:7)
```

The oak code is attempting a [`.truncate` on a `Deno.FsFile`](https://github.com/oakserver/oak/blob/c4fe8b35c0362b9b153e1cf8af4081c7a3905834/multipart.ts#L253) and the op side of the code is:

https://github.com/denoland/deno/blob/1f7dd5eda9b5e17ff102abb4dce1f19a4ce76688/runtime/js/30_fs.js#L258

The error appears to be from the Rust side though, when trying to coerce the len.  For JavaScript it shouldn't be a problem handing file lengths of around 28_000_000_000 bytes, but I noticed the op arguments are:

https://github.com/denoland/deno/blob/1f7dd5eda9b5e17ff102abb4dce1f19a4ce76688/runtime/ops/fs.rs#L1519-L1524

Which seems like a problem for file lengths, as a) it is signed in a situation where signed isn't import and b) would limit file operations to 2_147_483_647, quite a lot smaller than a lot of files you find in the wild these days.

Is there any issue with using an `i64`, as that is the closest to what is supported natively in JavaScript (obviously JSON only supports i32, but we aren't using JSON in our ops anymore)?

**Comment by JetLua** [comment]
Will it be fixed in 1.22?

**Comment by lucacasonato** [comment]
This was fixed.
